live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/domain/event/IngestNewNodeEventHandler.java

```diff
@@ -26,39 +26,45 @@
+import static org.alfresco.hxi_connector.live_ingester.domain.model.out.PredefinedNodeProperty.IS_FOLDER;
+import static org.alfresco.hxi_connector.live_ingester.domain.model.out.PredefinedNodeProperty.MODIFIED_BY_USER_WITH_ID;
+import static org.alfresco.hxi_connector.live_ingester.domain.model.out.PredefinedNodeProperty.NAME;
+import static org.alfresco.hxi_connector.live_ingester.domain.model.out.PredefinedNodeProperty.PRIMARY_ASSOC_Q_NAME;
+import static org.alfresco.hxi_connector.live_ingester.domain.model.out.PredefinedNodeProperty.TYPE;
 
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.alfresco.hxi_connector.live_ingester.domain.model.in.IngestNewNodeEvent;
 import org.alfresco.hxi_connector.live_ingester.domain.model.in.Node;
 import org.alfresco.hxi_connector.live_ingester.domain.model.out.EventPublisher;
 import org.alfresco.hxi_connector.live_ingester.domain.model.out.event.UpdateNodeMetadataEvent;
+import org.alfresco.hxi_connector.live_ingester.domain.model.transform.request.TransformRequest;
+import org.alfresco.hxi_connector.live_ingester.domain.model.transform.request.TransformRequester;
 import org.springframework.stereotype.Component;
 
 @Component
+@Slf4j
 @RequiredArgsConstructor
 public class IngestNewNodeEventHandler
 {
-
+    private final UpdateNodeEventMapper updateNodeEventMapper;
     private final EventPublisher eventPublisher;
+    private final TransformRequestMapper transformRequestMapper;
+    private final TransformRequester transformRequester;
 
     public void handle(IngestNewNodeEvent event)
     {
-        Node node = event.node();
-
-        UpdateNodeMetadataEvent updateMetadataEvent = UpdateNodeMetadataEvent.create()
-                                                          .set(NAME.withValue(node.name()))
-                                                          .set(PRIMARY_ASSOC_Q_NAME.withValue(node.primaryAssocQName()))
-                                                          .set(TYPE.withValue(node.nodeType()))
-                                                          .set(CREATED_BY_USER_WITH_ID.withValue(node.createdByUserWithId()))
-                                                          .set(MODIFIED_BY_USER_WITH_ID.withValue(node.modifiedByUserWithId()))
-                                                          .set(ASPECTS_NAMES.withValue(node.aspectNames()))
-                                                          .set(IS_FILE.withValue(node.isFile()))
-                                                          .set(IS_FOLDER.withValue(node.isFolder()))
-                                                          .set(CREATED_AT.withValue(node.createdAt()));
-
-        node.properties().forEach(updateMetadataEvent::set);
-
+        UpdateNodeMetadataEvent updateMetadataEvent = updateNodeEventMapper.map(event);
         eventPublisher.publishMessage(updateMetadataEvent);
+
+        if (event.node().contentMimeType() != null)
```

Comment:
Yep - we can use an Optional for this.

Reply:
but what contentMimeType = null means?

Reply:
because maybe we could create ContentMimeType class and map null to some constant ContentMimeType.NO_CONTENT and everything else to ContentMimeType("some_type")